### DISCUSSION TITLE
Spotify activation not possible after last update 20211114

### DISCUSSION THREAD

**Discussion by Sigmun** [rern/rAudio#628] [Fixed Issues]
After last update 20211114, spotify renderer has changed and I need to provide a Client ID and Secret from the Spotify for Developers website, which I get.

But as soon as I paste one of them in their correspondig field, the blue OK button turn to black and I can't validate. Thats a pity because it means that I can't listen to spotify now.

I've also a doubt on the behaviour of this new spotify plugin: will it still be possible to stream from different accounts and devices as we use to do until now?

**Comment by rern** [comment]
Try the latest update.

**Comment by Raitsa** [comment]
I have the same kinda problem with the very latest version.
After inserting the client ID and secret all ok but after the redirect back to my rAudio, Spotify option turns off automatically in the rAudio features page. 

rAudio 1 • 20211126
5.10.74-1 • aarch64
20211004-1
Raspberry Pi 4 Model B Rev 1.4

**Comment by Raitsa** [comment]
Strange, with a tablet I managed to set client id and secret key and the stick. With the PC I did not manage to put those in.

Still have to figure on how to play anything from Spotify...

**Comment by Raitsa** [comment]
My prob with Spotify might be this false here:

spotifyd": true,
  "spotifydactive": false,
  "spotifyredirect": "https://rern.github.io/raudio/spotify",
  "spotifytoken": true,

Any idea how to set that to true? I have no idea why its on false.

**Comment by Raitsa** [comment]
After a lot of fiddling, I got Spotify Connect to work from the PC and then from the phones and tablets too. So all ok for me now..

**Reply by digixmax** [reply]
I am having exactly the same problem you had -- the "enable" spotify slider would revert to off after a short while as if nothing had ever been configured.

What fiddling did you do to get it to work?

**Reply by rern** [reply]
Try:
- Settings > Addons > rAudio - Update
- Enable Spotify
- 
If still no:
- Prepare an SD card with the latest image file
- Settings > Addons > rAudio - Update
- Enable Spotify

Previously, Spotify on rAudio use the same key to get playing status from Spotify server. There's a certain limit of requests for a key which will be hit sooner or later. One private key per one user makes sure this limit will never reached. Besides, current playing progress can be fetched directly without a lot of workaround.

**Reply by digixmax** [reply]
Thanks, after applying the update to v20211203 spotifyd configuration now persists.

**Reply by digixmax** [reply]
Just a quick note -- rAudio web interface didn't let me configure the App ID and Client Secret using Firefox -> I had to switch to Chrome to do so.

**Reply by aholloway17** [reply]
Sorry, I am a little late to this thread. I am trying to find the latest image for Rune, the newest one from the site is dated 2016. Is there a later image?

Cheers,

**Reply by digixmax** [reply]
> Sorry, I am a little late to this thread. I am trying to find the latest image for Rune, the newest one from the site is dated 2016. Is there a later image?
> 
> Cheers,

https://github.com/rern/rAudio-1/releases/tag/i20220214
